### PR TITLE
Use openlibm for GitHub Actions builds

### PR DESCRIPTION
**Pull request by willpett**
Sometimes, the FBD integration tests were failing on some platforms but not others, leading to speculation that different versions of the boost RNG were to blame (see #139). Actually, the discrepancies were caused by different implementations of the standard math library `libm`, causing minute differences in the output of `std::exp`. This modifies the GitHub Actions build so that all platforms use a common implementation, [openlibm](https://openlibm.org/).

I also added a couple other compiler flags (`-mfpmath=sse -ffp-contract=off`) to the integration tests to help ensure more deterministic floating point math across platforms (especially 32-bit systems).

Finally, a few small fixes to the build script make sure that all platform-specific artifacts are properly uploaded.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -117,15 +117,15 @@ jobs:
...
-          brew install pkg-config pandoc boost ccache coreutils
+          brew install pkg-config pandoc boost ccache coreutils openlibm
           sudo xcode-select -switch /Applications/Xcode_${{ matrix.version }}.app
           ccache --set-config=cache_dir=$HOME/.ccache
 
     - name: Install (Windows meson cross compile)
       if: matrix.name == 'windows'
       run: |
           sudo apt-get update
-          sudo apt-get install -y pandoc libcairo2-dev ccache dos2unix zstd
```
Comment (bredelings) [inline .github/workflows/build.yml:128]: I would think that we don't need this for a cross-compile?

**Review by bredelings — APPROVED** [review]
This looks good!

File: .github/workflows/build.yml (hunk 1 above)
Comment (willpett) [reply .github/workflows/build.yml:128]: I don't think there's an alternative package without them.

Comment (willpett) [reply .github/workflows/build.yml:128]: oh wait nevermind, I see what you're saying.

**Pull request merged**